custom/icds_reports/filters.py

```diff
@@ -35,40 +35,66 @@ def loc_to_json(loc, project):
             'can_edit': True
         }
 
-    project = Domain.get_by_name(domain)
+    def _get_accessible_location_ids(primary_location):
+        accessible_location_ids = set([primary_location.location_id])
+        accessible_location_ids.update(primary_location.get_ancestors().values_list('location_id', flat=True))
+        accessible_location_ids.update(primary_location.get_descendants().values_list('location_id', flat=True))
+        return accessible_location_ids
+
+    def _get_ancestor_loc_dict(location_list, location):
+        for parent_location in location_list:
+            if parent_location['uuid'] == location.location_id:
+                return parent_location
+        return None
+
+    user_location = user.get_sql_location(domain)
+    user_has_all_location_access = user.has_permission(domain, 'access_all_locations')
+
+    if user_has_all_location_access:
+        accessible_root_locations = SQLLocation.root_locations(domain)
```

Comment:
Since we just need the root locations if there is no selected location id, this is just trying to start with that. `accessible_location_ids` is called later only if it is needed. Does that make sense?

Reply:
Agree with Manish.

Reply:
that makes sense